--- sip/QtSvg/QtSvgmod.sip.old	2012-10-13 21:16:20.000000000 +0200
+++ sip/QtSvg/QtSvgmod.sip	2012-10-13 21:16:20.000000000 +0200
@@ -56,7 +56,15 @@
 
 %DefaultSupertype sip.simplewrapper
 
-%Include qgraphicssvgitem.sip
-%Include qsvggenerator.sip
-%Include qsvgrenderer.sip
-%Include qsvgwidget.sip
+%If (PyQt_GraphicsSVGItem)
+  %Include qgraphicssvgitem.sip
+%End
+%If (PyQt_SVGGenerator)
+  %Include qsvggenerator.sip
+%End
+%If (PyQt_SVGRenderer)
+  %Include qsvgrenderer.sip
+%End
+%If (PyQt_SVGWidget)
+  %Include qsvgwidget.sip
+%End
