--- sip/phonon/phononmod.sip.old	2012-10-13 21:16:20.000000000 +0200
+++ sip/phonon/phononmod.sip	2012-10-13 21:16:20.000000000 +0200
@@ -62,7 +62,9 @@
 %Include backendcapabilities.sip
 %Include effect.sip
 %Include effectparameter.sip
-%Include effectwidget.sip
+%If (PyQt_Phonon_EffectWidget)
+  %Include effectwidget.sip
+%End
 %Include mediacontroller.sip
 %Include medianode.sip
 %Include mediaobject.sip
@@ -70,7 +72,11 @@
 %Include objectdescription.sip
 %Include path.sip
 %Include phononnamespace.sip
-%Include seekslider.sip
+%If (PyQt_Phonon_SeekSlider)
+  %Include seekslider.sip
+%End
 %Include videoplayer.sip
 %Include videowidget.sip
-%Include volumeslider.sip
+%If (PyQt_Phonon_VolumeSlider)
+  %Include volumeslider.sip
+%End
