--- qpy/QtCore/qpycore_qstring.cpp.old	2012-10-13 21:16:19.000000000 +0200
+++ qpy/QtCore/qpycore_qstring.cpp	2012-10-13 21:16:19.000000000 +0200
@@ -26,8 +26,11 @@
 #include <Python.h>
 #include <string.h>
 
+#include <qconfig.h>
 #include <QString>
+#ifndef QT_NO_TEXTCODEC
 #include <QTextCodec>
+#endif
 #include <QVector>
 
 #include "qpycore_sip.h"
@@ -165,6 +168,7 @@
         }
         else
         {
+#ifndef QT_NO_TEXTCODEC
             QTextCodec *codec = QTextCodec::codecForTr();
 
             if (codec)
@@ -184,6 +188,9 @@
             {
                 obj = PyUnicode_AsLatin1String(obj);
             }
+#else
+                obj = PyUnicode_AsLatin1String(obj);
+#endif
         }
 
         if (obj)
