--- sip/QtNetwork/qabstractnetworkcache.sip.old	2012-10-13 21:16:20.000000000 +0200
+++ sip/QtNetwork/qabstractnetworkcache.sip	2012-10-13 21:16:20.000000000 +0200
@@ -173,12 +173,14 @@
 };
 
 %End
+%If (PyQt_DataStream)
 %If (Qt_4_5_0 -)
 QDataStream &operator<<(QDataStream &, const QNetworkCacheMetaData & /Constrained/);
 %End
 %If (Qt_4_5_0 -)
 QDataStream &operator>>(QDataStream &, QNetworkCacheMetaData & /Constrained/);
 %End
+%End
 %If (Qt_4_5_0 -)
 
 class QAbstractNetworkCache : QObject
