--- sip/QtCore/qbitarray.sip.old	2012-10-13 21:16:20.000000000 +0200
+++ sip/QtCore/qbitarray.sip	2012-10-13 21:16:20.000000000 +0200
@@ -93,8 +93,10 @@
 QBitArray operator&(const QBitArray &, const QBitArray &);
 QBitArray operator|(const QBitArray &, const QBitArray &);
 QBitArray operator^(const QBitArray &, const QBitArray &);
+%If (PyQt_DataStream)
 QDataStream &operator<<(QDataStream &, const QBitArray & /Constrained/);
 QDataStream &operator>>(QDataStream &, QBitArray & /Constrained/);
+%End
 %If (Qt_4_3_0 -)
 void qSwap(QBitArray &value1, QBitArray &value2);
 %End
