--- sip/QtCore/qchar.sip.old	2012-10-13 21:16:20.000000000 +0200
+++ sip/QtCore/qchar.sip	2012-10-13 21:16:20.000000000 +0200
@@ -423,6 +423,7 @@
 bool operator>=(QChar c1, QChar c2);
 bool operator<(QChar c1, QChar c2);
 bool operator>(QChar c1, QChar c2);
+%If (PyQt_DataStream)
 %If (Qt_5_0_0 -)
 QDataStream &operator<<(QDataStream &, QChar /Constrained/);
 %End
@@ -430,3 +431,4 @@
 QDataStream &operator<<(QDataStream &, const QChar & /Constrained/);
 %End
 QDataStream &operator>>(QDataStream &, QChar & /Constrained/);
+%End
