--- sip/QtCore/qcoreapplication.sip.old	2012-10-13 21:16:20.000000000 +0200
+++ sip/QtCore/qcoreapplication.sip	2012-10-13 21:16:20.000000000 +0200
@@ -186,12 +186,16 @@
     static bool closingDown();
     static QString applicationDirPath();
     static QString applicationFilePath();
+%If (PyQt_Library)
     static void setLibraryPaths(const QStringList &);
     static QStringList libraryPaths();
     static void addLibraryPath(const QString &);
     static void removeLibraryPath(const QString &);
+%End
+%If (PyQt_Translation)
     static void installTranslator(QTranslator *);
     static void removeTranslator(QTranslator *);
+%End
 
     enum Encoding
     {
@@ -403,18 +407,23 @@
     }
 }
 %End
+%If (PyQt_Thread)
 void pyqtRemoveInputHook();
 %MethodCode
     // Clear the Python input hook installed when the module was initialised.
     PyOS_InputHook = 0;
 %End
+%End
 
+%If (PyQt_Thread)
 void pyqtRestoreInputHook();
 %MethodCode
     // Restore the input hook.
     PyOS_InputHook = qtcore_input_hook;
 %End
+%End
 
+%If (PyQt_Thread)
 %ModuleCode
 #include <QCoreApplication>
 #include <QThread>
@@ -460,8 +469,11 @@
     return 0;
 }
 %End
+%End
 
+%If (PyQt_Thread)
 %PostInitialisationCode
 // Process events from the input hook.
 PyOS_InputHook = qtcore_input_hook;
 %End
+%End
