--- sip/QtCore/qcoreevent.sip.old	2012-10-13 21:16:20.000000000 +0200
+++ sip/QtCore/qcoreevent.sip	2012-10-13 21:16:20.000000000 +0200
@@ -50,7 +50,7 @@
             break;
     #endif
     
-    #if QT_VERSION >= 0x040600
+    #if QT_VERSION >= 0x040600 && defined(SIP_FEATURE_PyQt_StateMachine)
     case QEvent::StateMachineSignal:
             sipType = sipType_QStateMachine_SignalEvent;
             break;
@@ -275,12 +275,14 @@
 %If (Qt_4_6_0 -)
         WinIdChange,
 %End
+%If (PyQt_Gestures)
 %If (Qt_4_6_0 -)
         Gesture,
 %End
 %If (Qt_4_6_0 -)
         GestureOverride,
 %End
+%End
         User,
         MaxUser,
 %If (Qt_5_0_0 -)
