--- sip/QtCore/qnamespace.sip.old	2012-10-04 17:51:55.056239142 +0200
+++ sip/QtCore/qnamespace.sip	2012-10-04 17:51:55.060239142 +0200
@@ -1507,7 +1507,7 @@
     };
 
     typedef QFlags<Qt::MatchFlag> MatchFlags;
-%If (WS_WIN || WS_MACX)
+%If (WS_WIN || WS_MACX || WS_QWS)
     typedef void *HANDLE;
 %End
 %If (WS_X11)
@@ -1729,6 +1729,7 @@
 %If (Qt_4_6_0 -)
     typedef QFlags<Qt::TouchPointState> TouchPointStates;
 %End
+%If (PyQt_Gestures)
 %If (Qt_4_6_0 -)
 
     enum GestureState
@@ -1768,6 +1769,7 @@
 %If (Qt_4_6_0 -)
     typedef QFlags<Qt::GestureFlag> GestureFlags;
 %End
+%End
 %If (Qt_4_6_0 -)
 
     enum NavigationMode
@@ -1992,5 +1994,7 @@
 QFlags<Qt::TouchPointState> operator|(Qt::TouchPointState f1, QFlags<Qt::TouchPointState> f2);
 %End
 %If (Qt_4_6_0 -)
+%If (PyQt_Gestures)
 QFlags<Qt::GestureFlag> operator|(Qt::GestureFlag f1, QFlags<Qt::GestureFlag> f2);
 %End
+%End
