--- sip/QtCore/qsize.sip.old	2012-10-13 21:16:20.000000000 +0200
+++ sip/QtCore/qsize.sip	2012-10-13 21:16:20.000000000 +0200
@@ -96,8 +96,10 @@
 %End
 };
 
+%If (PyQt_DataStream)
 QDataStream &operator<<(QDataStream &, const QSize & /Constrained/);
 QDataStream &operator>>(QDataStream &, QSize & /Constrained/);
+%End
 bool operator==(const QSize &s1, const QSize &s2);
 bool operator!=(const QSize &s1, const QSize &s2);
 const QSize operator+(const QSize &s1, const QSize &s2);
