--- sip/QtCore/qstate.sip.old	2012-10-13 21:16:19.000000000 +0200
+++ sip/QtCore/qstate.sip	2012-10-13 21:16:19.000000000 +0200
@@ -73,7 +73,9 @@
     void setInitialState(QAbstractState *state /KeepReference/);
     QState::ChildMode childMode() const;
     void setChildMode(QState::ChildMode mode);
+%If (PyQt_Properties)
     void assignProperty(QObject *object, const char *name, const QVariant &value);
+%End
 
 signals:
     void finished();
