--- sip/QtCore/qurl.sip.old	2012-10-04 17:51:54.864239143 +0200
+++ sip/QtCore/qurl.sip	2012-10-04 17:51:54.872239143 +0200
@@ -249,8 +249,10 @@
 };
 
 QFlags<QUrl::FormattingOption> operator|(QUrl::FormattingOption f1, QFlags<QUrl::FormattingOption> f2);
+%If (PyQt_DataStream)
 QDataStream &operator<<(QDataStream &, const QUrl & /Constrained/);
 QDataStream &operator>>(QDataStream &, QUrl & /Constrained/);
+%End
 %If (Qt_4_3_0 -)
 void qSwap(QUrl &value1, QUrl &value2);
 %End
