Index: python-qt4-4.9.1/configure.py
===================================================================
--- python-qt4-4.9.1.orig/configure.py	2012-02-11 17:02:46.626548495 -0500
+++ python-qt4-4.9.1/configure.py	2012-02-11 21:58:02.088063515 -0500
@@ -870,31 +870,6 @@
             makefile.generate()
             tool.append("dbus")
 
-        # Only include ElementTree for older versions of Python.
-        if sipcfg.py_version < 0x020500:
-            sipconfig.inform("Creating elementtree Makefile...")
-
-            makefile = sipconfig.PythonModuleMakefile(
-                configuration=sipcfg,
-                dstdir=os.path.join(pyqt_modroot, "elementtree"),
-                dir="elementtree"
-            )
-
-            makefile.generate()
-            tool.append("elementtree")
-
-        # Create the pyuic4 wrapper.  Use the GUI version on MacOS (so that
-        # previews work properly and normal console use will work anyway), but
-        # not on Windows (so that normal console use will work).
-        sipconfig.inform("Creating pyuic4 wrapper...")
-
-        if sys.platform == 'darwin':
-            gui = True
-            use_arch = opts.use_arch
-        else:
-            gui = False
-            use_arch = ''
-
         # The pyuic directory may not exist if we are building away from the
         # source directory.
         try:
@@ -903,7 +878,6 @@
             pass
 
         uicdir=os.path.join(pyqt_modroot, "uic")
-        wrapper = sipconfig.create_wrapper(os.path.join(uicdir, "pyuic.py"), os.path.join("pyuic", "pyuic4"), gui, use_arch)
 
         sipconfig.inform("Creating pyuic4 Makefile...")
 
@@ -912,7 +886,6 @@
             dstdir=uicdir,
             srcdir=os.path.join(src_dir, "pyuic", "uic"),
             dir="pyuic",
-            installs=[[os.path.basename(wrapper), opts.pyqtbindir]]
         )
 
         makefile.generate()
