--- sip/QtGui/qabstractslider.sip.old	2012-10-13 21:16:21.000000000 +0200
+++ sip/QtGui/qabstractslider.sip	2012-10-13 21:16:21.000000000 +0200
@@ -98,7 +98,9 @@
     virtual bool event(QEvent *e);
     virtual void keyPressEvent(QKeyEvent *ev);
     virtual void timerEvent(QTimerEvent *);
+%If (PyQt_WheelEvent)
     virtual void wheelEvent(QWheelEvent *e);
+%End
     virtual void changeEvent(QEvent *e);
 
 private:
