--- sip/QtGui/qabstractbutton.sip.old	2012-10-04 17:51:54.340239144 +0200
+++ sip/QtGui/qabstractbutton.sip	2012-10-04 17:51:54.348239144 +0200
@@ -51,8 +51,10 @@
     void setIcon(const QIcon &icon);
     QIcon icon() const;
     QSize iconSize() const;
+%If (PyQt_Shortcut)
     void setShortcut(const QKeySequence &key);
     QKeySequence shortcut() const;
+%End
     void setCheckable(bool);
     bool isCheckable() const;
     bool isChecked() const;
@@ -62,7 +64,9 @@
     bool autoRepeat() const;
     void setAutoExclusive(bool);
     bool autoExclusive() const;
+%If (PyQt_ButtonGroup)
     QButtonGroup *group() const;
+%End
 
 public slots:
     void setIconSize(const QSize &size);
