--- sip/QtGui/qbrush.sip.old	2012-10-13 21:16:20.000000000 +0200
+++ sip/QtGui/qbrush.sip	2012-10-13 21:16:20.000000000 +0200
@@ -140,8 +140,10 @@
 %End
 };
 
+%If (PyQt_DataStream)
 QDataStream &operator>>(QDataStream &, QBrush & /Constrained/);
 QDataStream &operator<<(QDataStream &, const QBrush & /Constrained/);
+%End
 %If (Qt_5_0_0 -)
 typedef QVector<QPair<double, QColor>> QGradientStops;
 %End
