--- sip/QtGui/qdialog.sip.old	2012-10-13 21:16:20.000000000 +0200
+++ sip/QtGui/qdialog.sip	2012-10-13 21:16:20.000000000 +0200
@@ -108,7 +108,9 @@
     virtual void closeEvent(QCloseEvent *);
     virtual void showEvent(QShowEvent *);
     virtual void resizeEvent(QResizeEvent *);
+%If (PyQt_ContextMenu)
     virtual void contextMenuEvent(QContextMenuEvent *);
+%End
     virtual bool eventFilter(QObject *, QEvent *);
 
 private:
