--- sip/QtGui/qgraphicsproxywidget.sip.old	2012-10-13 21:16:21.000000000 +0200
+++ sip/QtGui/qgraphicsproxywidget.sip	2012-10-13 21:16:21.000000000 +0200
@@ -69,7 +69,9 @@
     virtual bool eventFilter(QObject *object, QEvent *event);
     virtual void showEvent(QShowEvent *event);
     virtual void hideEvent(QHideEvent *event);
+%If (PyQt_ContextMenu)
     virtual void contextMenuEvent(QGraphicsSceneContextMenuEvent *event);
+%End
     virtual void hoverEnterEvent(QGraphicsSceneHoverEvent *event);
     virtual void hoverLeaveEvent(QGraphicsSceneHoverEvent *event);
     virtual void hoverMoveEvent(QGraphicsSceneHoverEvent *event);
@@ -79,7 +81,9 @@
     virtual void mousePressEvent(QGraphicsSceneMouseEvent *event);
     virtual void mouseReleaseEvent(QGraphicsSceneMouseEvent *event);
     virtual void mouseDoubleClickEvent(QGraphicsSceneMouseEvent *event);
+%If (PyQt_WheelEvent)
     virtual void wheelEvent(QGraphicsSceneWheelEvent *event);
+%End
     virtual void keyPressEvent(QKeyEvent *event);
     virtual void keyReleaseEvent(QKeyEvent *event);
     virtual void focusInEvent(QFocusEvent *event);
@@ -87,6 +91,7 @@
     virtual bool focusNextPrevChild(bool next);
     virtual QSizeF sizeHint(Qt::SizeHint which, const QSizeF &constraint = QSizeF()) const;
     virtual void resizeEvent(QGraphicsSceneResizeEvent *event);
+%If (PyQt_DragAndDrop)
 %If (Qt_4_5_0 -)
     virtual void dragEnterEvent(QGraphicsSceneDragDropEvent *event);
 %End
@@ -99,6 +104,7 @@
 %If (Qt_4_5_0 -)
     virtual void dropEvent(QGraphicsSceneDragDropEvent *event);
 %End
+%End
 %If (Qt_4_5_0 -)
     QGraphicsProxyWidget *newProxyWidget(const QWidget *) /Factory/;
 %End
