--- sip/QtGui/qgraphicsview.sip.old	2012-10-13 21:16:21.000000000 +0200
+++ sip/QtGui/qgraphicsview.sip	2012-10-13 21:16:21.000000000 +0200
@@ -142,11 +142,15 @@
 protected:
     virtual bool event(QEvent *event);
     virtual bool viewportEvent(QEvent *event);
+%If (PyQt_ContextMenu)
     virtual void contextMenuEvent(QContextMenuEvent *event);
+%End
+%If (PyQt_DragAndDrop)
     virtual void dragEnterEvent(QDragEnterEvent *event);
     virtual void dragLeaveEvent(QDragLeaveEvent *event);
     virtual void dragMoveEvent(QDragMoveEvent *event);
     virtual void dropEvent(QDropEvent *event);
+%End
     virtual void focusInEvent(QFocusEvent *event);
     virtual void focusOutEvent(QFocusEvent *event);
 %If (Qt_4_4_0 -)
@@ -158,7 +162,9 @@
     virtual void mousePressEvent(QMouseEvent *event);
     virtual void mouseMoveEvent(QMouseEvent *event);
     virtual void mouseReleaseEvent(QMouseEvent *event);
+%If (PyQt_WheelEvent)
     virtual void wheelEvent(QWheelEvent *event);
+%End
     virtual void paintEvent(QPaintEvent *event);
     virtual void resizeEvent(QResizeEvent *event);
     virtual void scrollContentsBy(int dx, int dy);
@@ -310,12 +316,14 @@
 %If (Qt_4_3_0 -)
     void setOptimizationFlags(QFlags<QGraphicsView::OptimizationFlag> flags);
 %End
+%If (PyQt_RubberBand)
 %If (Qt_4_3_0 -)
     Qt::ItemSelectionMode rubberBandSelectionMode() const;
 %End
 %If (Qt_4_3_0 -)
     void setRubberBandSelectionMode(Qt::ItemSelectionMode mode);
 %End
+%End
 %If (Qt_4_3_0 -)
     QTransform transform() const;
 %End
