--- sip/QtGui/qiconengine.sip.old	2012-10-13 21:16:21.000000000 +0200
+++ sip/QtGui/qiconengine.sip	2012-10-13 21:16:21.000000000 +0200
@@ -92,8 +92,10 @@
 public:
     virtual QString key() const;
     virtual QIconEngineV2 *clone() const /Factory/;
+%If (PyQt_DataStream)
     virtual bool read(QDataStream &in);
     virtual bool write(QDataStream &out) const;
+%End
 %If (Qt_4_5_0 -)
 
     enum IconEngineHook
