--- sip/QtGui/qinputcontext.sip.old	2012-10-13 21:16:21.000000000 +0200
+++ sip/QtGui/qinputcontext.sip	2012-10-13 21:16:21.000000000 +0200
@@ -46,7 +46,9 @@
     QWidget *focusWidget() const;
     virtual void setFocusWidget(QWidget *w);
     virtual void widgetDestroyed(QWidget *w);
+%If (PyQt_Action)
     virtual QList<QAction*> actions();
+%End
     virtual bool filterEvent(const QEvent *event);
     void sendEvent(const QInputMethodEvent &event);
 
