--- sip/QtGui/qpixmap.sip.old	2012-10-13 21:16:21.000000000 +0200
+++ sip/QtGui/qpixmap.sip	2012-10-13 21:16:21.000000000 +0200
@@ -40,6 +40,7 @@
     QPixmap(int w, int h);
     QPixmap(const QSize &);
     QPixmap(const QString &fileName, const char *format = 0, Qt::ImageConversionFlags flags = Qt::AutoColor);
+%If (PyQt_ImageFormat_XPM)
     QPixmap(SIP_PYLIST xpm /DocType="list-of-str"/) [(const char **xpm)];
 %MethodCode
         // The Python interface is a list of strings that make up the image.
@@ -54,6 +55,7 @@
         else
             sipIsErr = 1;
 %End
+%End
 
     QPixmap(const QPixmap &);
     QPixmap(const QVariant &variant) /NoDerived/;
@@ -89,7 +91,9 @@
     void setAlphaChannel(const QPixmap &);
     bool hasAlpha() const;
     bool hasAlphaChannel() const;
+%If (PyQt_Image_Heuristic_Mask)
     QBitmap createHeuristicMask(bool clipTight = true) const;
+%End
 %If (Qt_5_0_0 -)
     QBitmap createMaskFromColor(const QColor &maskColor, Qt::MaskMode mode = Qt::MaskInColor) const;
 %End
@@ -222,8 +226,10 @@
 %End
 };
 
+%If (PyQt_DataStream)
 QDataStream &operator<<(QDataStream &, const QPixmap & /Constrained/);
 QDataStream &operator>>(QDataStream &, QPixmap & /Constrained/);
+%End
 %If (Qt_4_3_0 -)
 void qSwap(QPixmap &value1, QPixmap &value2);
 %End
