--- sip/QtGui/qtextdocument.sip.old	2012-10-04 17:51:53.900239147 +0200
+++ sip/QtGui/qtextdocument.sip	2012-10-04 17:51:53.904239149 +0200
@@ -84,8 +84,10 @@
 
     void setMetaInformation(QTextDocument::MetaInformation info, const QString &);
     QString metaInformation(QTextDocument::MetaInformation info) const;
+%If (PyQt_TextHTMLParser)
     QString toHtml(const QByteArray &encoding = QByteArray()) const;
     void setHtml(const QString &html);
+%End
     QString toPlainText() const;
     void setPlainText(const QString &text);
 
@@ -183,12 +185,14 @@
 %If (Qt_4_2_0 -)
     int blockCount() const;
 %End
+%If (PyQt_CSSParser)
 %If (Qt_4_2_0 -)
     void setDefaultStyleSheet(const QString &sheet);
 %End
 %If (Qt_4_2_0 -)
     QString defaultStyleSheet() const;
 %End
+%End
 %If (Qt_4_2_0 -)
     void undo(QTextCursor *cursor);
 %End
