--- sip/QtGui/qtextformat.sip.old	2012-10-13 21:16:20.000000000 +0200
+++ sip/QtGui/qtextformat.sip	2012-10-13 21:16:20.000000000 +0200
@@ -56,8 +56,10 @@
 %End
 };
 
+%If (PyQt_DataStream)
 QDataStream &operator<<(QDataStream &, const QTextLength & /Constrained/);
 QDataStream &operator>>(QDataStream &, QTextLength & /Constrained/);
+%End
 
 class QTextFormat
 {
@@ -316,8 +318,10 @@
 %End
 };
 
+%If (PyQt_DataStream)
 QDataStream &operator<<(QDataStream &, const QTextFormat & /Constrained/);
 QDataStream &operator>>(QDataStream &, QTextFormat & /Constrained/);
+%End
 
 class QTextCharFormat : QTextFormat
 {
