--- sip/QtGui/qtextdocumentfragment.sip.old	2012-10-04 17:51:53.904239149 +0200
+++ sip/QtGui/qtextdocumentfragment.sip	2012-10-04 17:51:53.908239149 +0200
@@ -39,13 +39,17 @@
     ~QTextDocumentFragment();
     bool isEmpty() const;
     QString toPlainText() const;
+%If (PyQt_TextHTMLParser)
     QString toHtml() const;
 %If (Qt_4_2_0 -)
     QString toHtml(const QByteArray &encoding) const;
 %End
+%End
     static QTextDocumentFragment fromPlainText(const QString &plainText);
+%If (PyQt_TextHTMLParser)
     static QTextDocumentFragment fromHtml(const QString &html);
 %If (Qt_4_2_0 -)
     static QTextDocumentFragment fromHtml(const QString &html, const QTextDocument *resourceProvider);
 %End
+%End
 };
