--- sip/QtGui/qtextobject.sip.old	2012-10-13 21:16:21.000000000 +0200
+++ sip/QtGui/qtextobject.sip	2012-10-13 21:16:21.000000000 +0200
@@ -247,12 +247,14 @@
     QTextCharFormat charFormat() const;
     int charFormatIndex() const;
     QString text() const;
+%If (PyQt_RawFont)
 %If (Qt_5_0_0 -)
     QList<QGlyphRun> glyphRuns(int from = -1, int length = -1) const;
 %End
 %If (Qt_4_8_0 - Qt_5_0_0)
     QList<QGlyphRun> glyphRuns() const;
 %End
+%End
 };
 
 class QTextBlockUserData /Supertype=sip.wrapper/
