--- sip/QtXml/qdom.sip.old	2012-10-13 21:16:21.000000000 +0200
+++ sip/QtXml/qdom.sip	2012-10-13 21:16:21.000000000 +0200
@@ -141,7 +141,9 @@
     QDomProcessingInstruction toProcessingInstruction() const;
     QDomCharacterData toCharacterData() const;
     QDomComment toComment() const;
+%If (PyQt_TextStream)
     void save(QTextStream &, int) const /ReleaseGIL/;
+%End
     QDomElement firstChildElement(const QString &tagName /DocValue="Py_v3:''"/ = QString()) const;
     QDomElement lastChildElement(const QString &tagName /DocValue="Py_v3:''"/ = QString()) const;
     QDomElement previousSiblingElement(const QString &tagName /DocValue="Py_v3:''"/ = QString()) const;
@@ -158,8 +160,10 @@
 
 %End
 %If (Qt_4_3_0 -)
+%If (PyQt_TextStream)
     void save(QTextStream &, int, QDomNode::EncodingPolicy) const /ReleaseGIL/;
 %End
+%End
 };
 
 class QDomNodeList
@@ -451,4 +455,6 @@
     QDomNode::NodeType nodeType() const;
 };
 
+%If (PyQt_TextStream)
 QTextStream &operator<<(QTextStream &, const QDomNode &);
+%End
