--- sip/QtNetwork/qabstractsocket.sip.old	2012-10-13 21:16:20.000000000 +0200
+++ sip/QtNetwork/qabstractsocket.sip	2012-10-13 21:16:20.000000000 +0200
@@ -215,8 +215,10 @@
     virtual bool waitForReadyRead(int msecs = 30000) /ReleaseGIL/;
     virtual bool waitForBytesWritten(int msecs = 30000) /ReleaseGIL/;
     bool waitForDisconnected(int msecs = 30000) /ReleaseGIL/;
+%If (PyQt_NetworkProxy)
     void setProxy(const QNetworkProxy &networkProxy);
     QNetworkProxy proxy() const;
+%End
 
 signals:
     void hostFound();
@@ -225,8 +227,10 @@
     void stateChanged(QAbstractSocket::SocketState);
     void error(QAbstractSocket::SocketError);
 %If (Qt_4_3_0 -)
+%If (PyQt_NetworkProxy)
     void proxyAuthenticationRequired(const QNetworkProxy &proxy, QAuthenticator *authenticator);
 %End
+%End
 
 protected slots:
     void connectToHostImplementation(const QString &hostName, quint16 port, QFlags<QIODevice::OpenModeFlag> mode = QIODevice::ReadWrite);
