--- sip/QtCore/qabstractitemmodel.sip.old	2012-10-13 21:16:20.000000000 +0200
+++ sip/QtCore/qabstractitemmodel.sip	2012-10-13 21:16:20.000000000 +0200
@@ -214,8 +214,10 @@
 %End
 
 protected:
+%If (PyQt_DataStream)
     void encodeData(const QModelIndexList &indexes, QDataStream &stream) const;
     bool decodeData(int row, int column, const QModelIndex &parent, QDataStream &stream);
+%End
     void beginInsertRows(const QModelIndex &parent, int first, int last);
     void endInsertRows();
     void beginRemoveRows(const QModelIndex &parent, int first, int last);
