--- sip/QtCore/qstringlist.sip.old	2012-10-13 21:16:20.000000000 +0200
+++ sip/QtCore/qstringlist.sip	2012-10-13 21:16:20.000000000 +0200
@@ -268,5 +268,7 @@
 QStringList &operator+=(const QString &value);
 };
 
+%If (PyQt_DataStream)
 QDataStream &operator>>(QDataStream &in, QStringList &list /Constrained/) /API=QString: - 2/;
 QDataStream &operator<<(QDataStream &out, const QStringList &list /Constrained/) /API=QString: - 2/;
+%End
