--- sip/QtCore/quuid.sip.old	2012-10-13 21:16:20.000000000 +0200
+++ sip/QtCore/quuid.sip	2012-10-13 21:16:20.000000000 +0200
@@ -132,10 +132,13 @@
 
     QUuid();
     QUuid(uint l, ushort w1, ushort w2, uchar b1, uchar b2, uchar b3, uchar b4, uchar b5, uchar b6, uchar b7, uchar b8);
+%If (PyQt_UUIDString)
     QUuid(const QString &);
+%End
 %If (Qt_4_8_0 -)
     QUuid(const QByteArray &);
 %End
+%If (PyQt_UUIDString)
     SIP_PYOBJECT __repr__() const /DocType="str"/;
 %MethodCode
         PyObject *uni = qpycore_PyObject_FromQString(sipCpp->toString());
@@ -153,8 +156,11 @@
             Py_DECREF(uni);
         }
 %End
+%End
 
+%If (PyQt_UUIDString)
     QString toString() const;
+%End
     bool isNull() const;
     bool operator==(const QUuid &orig) const;
     bool operator!=(const QUuid &orig) const;
@@ -163,7 +169,9 @@
     static QUuid createUuid();
     QUuid::Variant variant() const;
     QUuid::Version version() const;
+%If (PyQt_UUIDString)
     operator QString() const;
+%End
 %If (Qt_4_8_0 -)
     QByteArray toByteArray() const;
 %End
@@ -176,5 +184,7 @@
 };
 
 %End
+%If (PyQt_DataStream)
 QDataStream &operator<<(QDataStream &, const QUuid & /Constrained/);
 QDataStream &operator>>(QDataStream &, QUuid & /Constrained/);
+%End
