--- sip/QtCore/qvariant.sip.old	2012-10-13 21:16:20.000000000 +0200
+++ sip/QtCore/qvariant.sip	2012-10-13 21:16:20.000000000 +0200
@@ -231,8 +231,10 @@
         sipRes = qpycore_PyObject_FromQVariant(*sipCpp);
 %End
 
+%If (PyQt_DataStream)
     void load(QDataStream &ds);
     void save(QDataStream &ds) const;
+%End
 %If (Qt_5_0_0 -)
     static const char *typeToName(int typeId);
 %End
@@ -392,11 +394,13 @@
 %If (Qt_4_5_0 -)
 typedef QHash<QString, QVariant> QVariantHash /DocType="Py_v3:dict-of-str-object"/;
 %End
+%If (PyQt_DataStream)
 QDataStream &operator>>(QDataStream &s, QVariant &p /Constrained/) /API=QVariant: - 2/;
 typedef QMap<QString, QVariant> QVariantMap /DocType="Py_v3:dict-of-str-object"/;
 QDataStream &operator<<(QDataStream &s, const QVariant &p /Constrained/) /API=QVariant: - 2/;
 QDataStream &operator>>(QDataStream &s, QVariant::Type &p /Constrained,In/);
 QDataStream &operator<<(QDataStream &s, const QVariant::Type p /Constrained/);
+%End
 %If (Qt_4_3_0 -)
 void qSwap(QVariant &value1, QVariant &value2) /API=QVariant: - 2/;
 %End
