--- sip/QtGui/qdockwidget.sip.old	2012-10-13 21:16:21.000000000 +0200
+++ sip/QtGui/qdockwidget.sip	2012-10-13 21:16:21.000000000 +0200
@@ -72,7 +72,9 @@
     void setAllowedAreas(Qt::DockWidgetAreas areas);
     Qt::DockWidgetAreas allowedAreas() const;
     bool isAreaAllowed(Qt::DockWidgetArea area) const;
+%If (PyQt_Action)
     QAction *toggleViewAction() const /Transfer/;
+%End
 %If (Qt_4_3_0 -)
     void setTitleBarWidget(QWidget *widget /Transfer/);
 %End
