--- sip/QtGui/qkeysequence.sip.old	2012-10-13 21:16:20.000000000 +0200
+++ sip/QtGui/qkeysequence.sip	2012-10-13 21:16:20.000000000 +0200
@@ -251,8 +251,10 @@
 %End
 };
 
+%If (PyQt_DataStream)
 QDataStream &operator<<(QDataStream &in, const QKeySequence &ks /Constrained/);
 QDataStream &operator>>(QDataStream &out, QKeySequence &ks /Constrained/);
+%End
 %If (Qt_4_3_0 -)
 void qSwap(QKeySequence &value1, QKeySequence &value2);
 %End
