--- sip/QtGui/qlcdnumber.sip.old	2012-10-13 21:16:21.000000000 +0200
+++ sip/QtGui/qlcdnumber.sip	2012-10-13 21:16:21.000000000 +0200
@@ -64,7 +64,7 @@
     void setMode(QLCDNumber::Mode);
     QLCDNumber::SegmentStyle segmentStyle() const;
     void setSegmentStyle(QLCDNumber::SegmentStyle);
-    double value() const;
+    qreal value() const;
     int intValue() const;
     virtual QSize sizeHint() const;
     void display(const QString &str);
