--- sip/QtGui/qmdisubwindow.sip.old	2012-10-13 21:16:21.000000000 +0200
+++ sip/QtGui/qmdisubwindow.sip	2012-10-13 21:16:21.000000000 +0200
@@ -77,6 +77,7 @@
     int keyboardSingleStep() const;
     void setKeyboardPageStep(int step);
     int keyboardPageStep() const;
+%If (PyQt_Menu)
     void setSystemMenu(QMenu *systemMenu /Transfer/);
 %MethodCode
         // We have to break the parent association on any existing menu.
@@ -96,6 +97,7 @@
 %End
 
     QMenu *systemMenu() const /Transfer/;
+%End
 %If (Qt_4_4_0 -)
     QMdiArea *mdiArea() const;
 %End
@@ -105,7 +107,9 @@
     void aboutToActivate();
 
 public slots:
+%If (PyQt_Menu)
     void showSystemMenu();
+%End
     void showShaded();
 
 protected:
@@ -125,7 +129,9 @@
     virtual void mouseReleaseEvent(QMouseEvent *mouseEvent);
     virtual void mouseMoveEvent(QMouseEvent *mouseEvent);
     virtual void keyPressEvent(QKeyEvent *keyEvent);
+%If (PyQt_ContextMenu)
     virtual void contextMenuEvent(QContextMenuEvent *contextMenuEvent);
+%End
     virtual void focusInEvent(QFocusEvent *focusInEvent);
     virtual void focusOutEvent(QFocusEvent *focusOutEvent);
     virtual void childEvent(QChildEvent *childEvent);
