--- sip/QtGui/qmessagebox.sip.old	2012-10-13 21:16:20.000000000 +0200
+++ sip/QtGui/qmessagebox.sip	2012-10-13 21:16:20.000000000 +0200
@@ -254,12 +254,14 @@
 %If (Qt_4_2_0 -)
     void setInformativeText(const QString &text);
 %End
+%If (PyQt_TextEdit)
 %If (Qt_4_2_0 -)
     QString detailedText() const;
 %End
 %If (Qt_4_2_0 -)
     void setDetailedText(const QString &text);
 %End
+%End
 %If (Qt_4_2_0 -)
     void setWindowTitle(const QString &title);
 %End
