--- sip/QtGui/qpicture.sip.old	2012-10-13 21:16:20.000000000 +0200
+++ sip/QtGui/qpicture.sip	2012-10-13 21:16:20.000000000 +0200
@@ -199,8 +199,10 @@
     QPictureIO(const QPictureIO &);
 };
 
+%If (PyQt_DataStream)
 QDataStream &operator<<(QDataStream &in, const QPicture &p /Constrained/);
 QDataStream &operator>>(QDataStream &in, QPicture &p /Constrained/);
+%End
 %If (Qt_4_3_0 -)
 void qSwap(QPicture &value1, QPicture &value2);
 %End
