--- sip/QtGui/qplaintextedit.sip.old	2012-10-13 21:16:21.000000000 +0200
+++ sip/QtGui/qplaintextedit.sip	2012-10-13 21:16:21.000000000 +0200
@@ -74,7 +74,9 @@
     QString toPlainText() const;
     void ensureCursorVisible();
     virtual QVariant loadResource(int type, const QUrl &name);
+%If (PyQt_ContextMenu)
     QMenu *createStandardContextMenu() /Factory/;
+%End
     QTextCursor cursorForPosition(const QPoint &pos) const;
     QRect cursorRect(const QTextCursor &cursor) const;
     QRect cursorRect() const;
@@ -110,9 +112,11 @@
 
 public slots:
     void setPlainText(const QString &text);
+%If (PyQt_Clipboard)
     void cut();
     void copy();
     void paste();
+%End
     void undo();
     void redo();
     void clear();
@@ -145,16 +149,22 @@
     virtual void mouseReleaseEvent(QMouseEvent *e);
     virtual void mouseDoubleClickEvent(QMouseEvent *e);
     virtual bool focusNextPrevChild(bool next);
+%If (PyQt_ContextMenu)
     virtual void contextMenuEvent(QContextMenuEvent *e);
+%End
+%If (PyQt_DragAndDrop)
     virtual void dragEnterEvent(QDragEnterEvent *e);
     virtual void dragLeaveEvent(QDragLeaveEvent *e);
     virtual void dragMoveEvent(QDragMoveEvent *e);
     virtual void dropEvent(QDropEvent *e);
+%End
     virtual void focusInEvent(QFocusEvent *e);
     virtual void focusOutEvent(QFocusEvent *e);
     virtual void showEvent(QShowEvent *);
     virtual void changeEvent(QEvent *e);
+%If (PyQt_WheelEvent)
     virtual void wheelEvent(QWheelEvent *e);
+%End
     virtual void inputMethodEvent(QInputMethodEvent *);
     virtual QVariant inputMethodQuery(Qt::InputMethodQuery property) const;
     virtual QMimeData *createMimeDataFromSelection() const /Factory/;
