--- sip/QtGui/qscrollbar.sip.old	2012-10-13 21:16:21.000000000 +0200
+++ sip/QtGui/qscrollbar.sip	2012-10-13 21:16:21.000000000 +0200
@@ -47,7 +47,9 @@
     virtual void mouseReleaseEvent(QMouseEvent *);
     virtual void mouseMoveEvent(QMouseEvent *);
     virtual void hideEvent(QHideEvent *);
+%If (PyQt_ContextMenu)
     virtual void contextMenuEvent(QContextMenuEvent *);
+%End
     virtual void sliderChange(QAbstractSlider::SliderChange change);
 
 private:
