--- sip/QtGui/qstyleoption.sip.old	2012-10-16 12:58:50.225280105 +0200
+++ sip/QtGui/qstyleoption.sip	2012-10-16 13:12:12.688691214 +0200
@@ -118,24 +118,31 @@
         break;
     #endif
     
+    #if defined(SIP_FEATURE_PyQt_RubberBand)
     case QStyleOption::SO_RubberBand:
         sipType = sipType_QStyleOptionRubberBand;
         break;
+    #endif
     
-    #if QT_VERSION >= 0x040200
+    #if QT_VERSION >= 0x040200 && defined(SIP_FEATURE_PyQt_SizeGrip)
     case QStyleOption::SO_SizeGrip:
         sipType = sipType_QStyleOptionSizeGrip;
         break;
     #endif
     
+    #if defined(SIP_FEATURE_PyQt_Slider)
     case QStyleOption::SO_Slider:
         sipType = sipType_QStyleOptionSlider;
         break;
+    #endif
     
+    #if defined(SIP_FEATURE_PyQt_SpinBox)
     case QStyleOption::SO_SpinBox:
         sipType = sipType_QStyleOptionSpinBox;
         break;
+    #endif
     
+    #if defined(SIP_FEATURE_PyQt_TabBar)
     case QStyleOption::SO_Tab:
     #if QT_VERSION >= 0x050000
         sipType = sipType_QStyleOptionTab;
@@ -179,7 +186,9 @@
         }
         break;
     #endif
+    #endif // SIP_FEATURE_PyQt_TabBar
     
+    #if defined(SIP_FEATURE_PyQt_TabWidget)
     case QStyleOption::SO_TabWidgetFrame:
     #if QT_VERSION >= 0x050000
         sipType = sipType_QStyleOptionTabWidgetFrame;
@@ -198,12 +207,13 @@
         }
         break;
     #endif
+    #endif // SIP_FEATURE_PyQt_TabWidget
     
     case QStyleOption::SO_TitleBar:
         sipType = sipType_QStyleOptionTitleBar;
         break;
     
-    #if QT_VERSION >= 0x040100
+    #if QT_VERSION >= 0x040100 && defined(SIP_FEATURE_PyQt_ToolBar)
     case QStyleOption::SO_ToolBar:
         sipType = sipType_QStyleOptionToolBar;
         break;
@@ -232,6 +242,7 @@
         sipType = sipType_QStyleOptionToolButton;
         break;
     
+    #if defined(SIP_FEATURE_PyQt_ItemViews)
     case QStyleOption::SO_ViewItem:
     #if QT_VERSION >= 0x050000
         sipType = sipType_QStyleOptionViewItem;
@@ -262,6 +273,7 @@
         }
         break;
     #endif
+    #endif // SIP_FEATURE_PyQt_ItemViews
     
     default:
         if ((sipCpp->type & QStyleOption::SO_ComplexCustomBase) == QStyleOption::SO_ComplexCustomBase)
@@ -509,6 +521,9 @@
 QFlags<QStyleOptionFrameV2::FrameFeature> operator|(QStyleOptionFrameV2::FrameFeature f1, QFlags<QStyleOptionFrameV2::FrameFeature> f2);
 %End
 
+%If (PyQt_TabBar)
+%If (PyQt_TabWidget)
+
 class QStyleOptionTabWidgetFrame : QStyleOption
 {
 %TypeHeaderCode
@@ -564,6 +579,8 @@
     QStyleOptionTabWidgetFrame(const QStyleOptionTabWidgetFrame &other);
 };
 
+%End // PyQt_TabWidget
+
 class QStyleOptionTabBarBase : QStyleOption
 {
 %TypeHeaderCode
@@ -613,6 +630,8 @@
     QStyleOptionTabBarBase(const QStyleOptionTabBarBase &other);
 };
 
+%End // PyQt_TabBar
+
 class QStyleOptionHeader : QStyleOption
 {
 %TypeHeaderCode
@@ -752,6 +771,8 @@
 
 QFlags<QStyleOptionButton::ButtonFeature> operator|(QStyleOptionButton::ButtonFeature f1, QFlags<QStyleOptionButton::ButtonFeature> f2);
 
+%If (PyQt_TabBar)
+
 class QStyleOptionTab : QStyleOption
 {
 %TypeHeaderCode
@@ -872,6 +893,8 @@
 
 %End
 
+%End
+
 class QStyleOptionProgressBar : QStyleOption
 {
 %TypeHeaderCode
@@ -1104,6 +1127,7 @@
     QStyleOptionDockWidget(const QStyleOptionDockWidget &other);
 };
 
+%If (PyQt_ItemViews)
 class QStyleOptionViewItem : QStyleOption
 {
 %TypeHeaderCode
@@ -1215,6 +1239,7 @@
 %If (Qt_5_0_0 -)
 QFlags<QStyleOptionViewItem::ViewItemFeature> operator|(QStyleOptionViewItem::ViewItemFeature f1, QFlags<QStyleOptionViewItem::ViewItemFeature> f2);
 %End
+%End // PyQt_ItemViews
 
 class QStyleOptionToolBox : QStyleOption
 {
@@ -1288,6 +1313,8 @@
     QStyleOptionToolBox(const QStyleOptionToolBox &other);
 };
 
+%If (PyQt_RubberBand)
+
 class QStyleOptionRubberBand : QStyleOption
 {
 %TypeHeaderCode
@@ -1333,6 +1360,8 @@
     QStyleOptionRubberBand(const QStyleOptionRubberBand &other);
 };
 
+%End
+
 class QStyleOptionComplex : QStyleOption
 {
 %TypeHeaderCode
@@ -1378,6 +1407,7 @@
     QStyleOptionComplex(const QStyleOptionComplex &other);
 };
 
+%If (PyQt_Slider)
 class QStyleOptionSlider : QStyleOptionComplex
 {
 %TypeHeaderCode
@@ -1432,7 +1462,9 @@
     QStyleOptionSlider();
     QStyleOptionSlider(const QStyleOptionSlider &other);
 };
+%End
 
+%If (PyQt_SpinBox)
 class QStyleOptionSpinBox : QStyleOptionComplex
 {
 %TypeHeaderCode
@@ -1478,6 +1510,7 @@
     QStyleOptionSpinBox();
     QStyleOptionSpinBox(const QStyleOptionSpinBox &other);
 };
+%End
 
 class QStyleOptionToolButton : QStyleOptionComplex
 {
@@ -1746,6 +1779,8 @@
     QRegion region;
 };
 
+%If (PyQt_ToolBar)
+
 class QStyleOptionToolBar : QStyleOption
 {
 %TypeHeaderCode
@@ -1813,6 +1848,8 @@
 
 QFlags<QStyleOptionToolBar::ToolBarFeature> operator|(QStyleOptionToolBar::ToolBarFeature f1, QFlags<QStyleOptionToolBar::ToolBarFeature> f2);
 
+%End
+
 class QStyleOptionGroupBox : QStyleOptionComplex
 {
 %TypeHeaderCode
@@ -1867,6 +1904,7 @@
     QStyleOptionGroupBox(const QStyleOptionGroupBox &other);
 };
 
+%If (PyQt_ItemViews)
 %If (Qt_4_2_0 - Qt_5_0_0)
 
 class QStyleOptionViewItemV2 : QStyleOptionViewItem
@@ -1908,7 +1946,10 @@
 %If (Qt_4_2_0 - Qt_5_0_0)
 QFlags<QStyleOptionViewItemV2::ViewItemFeature> operator|(QStyleOptionViewItemV2::ViewItemFeature f1, QFlags<QStyleOptionViewItemV2::ViewItemFeature> f2);
 %End
+%End // PyQt_ItemViews
+
 %If (Qt_4_2_0 -)
+%If (PyQt_SizeGrip)
 
 class QStyleOptionSizeGrip : QStyleOptionComplex
 {
@@ -1933,6 +1974,7 @@
 };
 
 %End
+%End
 %If (Qt_4_2_0 -)
 
 class QStyleOptionGraphicsItem : QStyleOption
@@ -1984,6 +2026,7 @@
 };
 
 %End
+%If (PyQt_ItemViews)
 %If (Qt_4_3_0 - Qt_5_0_0)
 
 class QStyleOptionViewItemV3 : QStyleOptionViewItemV2
@@ -2006,6 +2049,7 @@
 };
 
 %End
+%End // PyQt_ItemViews
 %If (Qt_4_3_0 - Qt_5_0_0)
 
 class QStyleOptionToolBoxV2 : QStyleOptionToolBox
@@ -2067,6 +2111,8 @@
 };
 
 %End
+
+%If (PyQt_ItemViews)
 %If (Qt_4_4_0 - Qt_5_0_0)
 
 class QStyleOptionViewItemV4 : QStyleOptionViewItemV3
@@ -2102,6 +2148,7 @@
 };
 
 %End
+%End // PyQt_ItemViews
 %If (Qt_4_5_0 - Qt_5_0_0)
 
 class QStyleOptionFrameV3 : QStyleOptionFrameV2
@@ -2124,6 +2171,8 @@
 };
 
 %End
+
+%If (PyQt_TabBar)
 %If (Qt_4_5_0 - Qt_5_0_0)
 
 class QStyleOptionTabBarBaseV2 : QStyleOptionTabBarBase
@@ -2169,6 +2218,9 @@
 };
 
 %End
+%End // PyQt_TabBar
+
+%If (PyQt_TabWidget)
 %If (Qt_4_6_0 - Qt_5_0_0)
 
 class QStyleOptionTabWidgetFrameV2 : QStyleOptionTabWidgetFrame
@@ -2191,3 +2243,4 @@
 };
 
 %End
+%End // PyQt_TabWidget
