--- sip/QtScript/qscriptcontextinfo.sip.old	2012-10-13 21:16:21.000000000 +0200
+++ sip/QtScript/qscriptcontextinfo.sip	2012-10-13 21:16:21.000000000 +0200
@@ -71,9 +71,11 @@
 %If (Qt_4_4_0 -)
 typedef QList<QScriptContextInfo> QScriptContextInfoList;
 %End
+%If (PyQt_DataStream)
 %If (Qt_4_4_0 -)
 QDataStream &operator<<(QDataStream &, const QScriptContextInfo & /Constrained/);
 %End
 %If (Qt_4_4_0 -)
 QDataStream &operator>>(QDataStream &, QScriptContextInfo & /Constrained/);
 %End
+%End
